src/Thoth.Json/Decode.fs

```diff
@@ -874,15 +874,28 @@ module Decode =
                 | Required -> false
 
     // TODO: Same API as for Thot.Json.Net.Decoder.BoxedDecoder
-    type private BoxedDecoder = Decoder<obj>
+    type BoxedDecoder = Decoder<obj>
+    type ExtraDecoders = Map<string, BoxedDecoder>
 
     // As generics are erased by Fable, let's just do an unsafe cast for performance
-    let inline private boxDecoder (d: Decoder<'T>): BoxedDecoder =
+    let inline boxDecoder (d: Decoder<'T>): BoxedDecoder =
         !!d // d >> Result.map box
 
-    let inline private unboxDecoder (d: BoxedDecoder): Decoder<'T> =
+    let inline unboxDecoder (d: BoxedDecoder): Decoder<'T> =
         !!d // d >> Result.map unbox
 
+    let inline makeExtra(): ExtraDecoders = Map.empty
+    let inline withInt64 (extra: ExtraDecoders): ExtraDecoders =
```

Comment:
Hum, I think we said that `Auto` features should be complete by default out of the box.

Isn't `extra` against this statement ?

People, who use `Auto` wants it to work magically by default for any types or perhaps I am wrong.